debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/j9/gc/GCObjectModel_V1.java

```diff
@@ -170,21 +170,38 @@ public UDATA getSizeInBytesWithHeader(J9ObjectPointer object) throws CorruptData
 		}
 	}
 
-@Override
-public UDATA getTotalFootprintInBytesWithHeader(J9ObjectPointer object) throws CorruptDataException {
-	if (isIndexable(object)) {
-		return adjustSizeInBytes(indexableObjectModel.getTotalFootprintInBytesWithHeader(J9IndexableObjectPointer.cast(object)));
-	} else {
-		return adjustSizeInBytes(mixedObjectModel.getSizeInBytesWithHeader(object));
+	@Override
+	public UDATA getTotalFootprintInBytesWithHeader(J9ObjectPointer object) throws CorruptDataException {
+		if (isIndexable(object)) {
+			return adjustSizeInBytes(indexableObjectModel.getTotalFootprintInBytesWithHeader(J9IndexableObjectPointer.cast(object)));
+		} else {
+			return adjustSizeInBytes(mixedObjectModel.getSizeInBytesWithHeader(object));
+		}
 	}
-}
 
-@Override
-	public boolean isDeadObject(J9ObjectPointer object) throws CorruptDataException 
+	@Override
+	public boolean isHoleObject(J9ObjectPointer object) throws CorruptDataException
 	{
 		return J9ObjectHelper.flags(object).allBitsIn(J9_GC_OBJ_HEAP_HOLE);
 	}
 
+	@Override
+	public boolean isDarkMatterObject(J9ObjectPointer object) throws CorruptDataException {
+		try {
+			if (J9ObjectHelper.clazz(object).classDepthAndFlags().anyBitsIn(J9AccClassDying)) {
+				// If an object's class's classDepthAndFlags includes J9AccClassDying,
+				// then the class has been unloaded and this object is left as dark matter
+				// for potential future clean up, so we can treat it as dead.
+				// See https://github.com/eclipse-openj9/openj9/issues/10537
+				return true;
+			}
+		} catch (CorruptDataException e) {
+			// If there was something wrong checking the object's class, then keep
+			// the old behavior in case further diagnostics can be done by the user.
```

Comment:
This comment could be improved: since this is a new method, there is no 'old behavior'.